src/main/scala/temple/test/internal/ServiceTestUtils.scala

```diff
@@ -0,0 +1,40 @@
+package temple.test.internal
+
+import io.circe.parser.parse
+import io.circe.{Json, JsonObject}
+import scalaj.http.Http
+import temple.utils.StringUtils
+
+object ServiceTestUtils {
+
+  /**
+    * Execute a POST request, where the response body is expected to be a JSON object
+    * @param url The URL to send the request to, prefixed with the protocol
+    * @param body The JSON body to send along with the request
+    * @param test The endpoint under test
+    * @return The decoded JSON response from the request
+    */
+  def postRequest(url: String, body: Json, test: EndpointTest): JsonObject = {
+    val response = Http(url).postData(body.toString).method("POST").asString
+    test.assertEqual(200, response.code)
+
+    parse(response.body) match {
```

Comment:
Would this block look better with fromEither and getOrElse, to avoid the nesting and to make it clearer where the return value is?